src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -1027,4 +1035,76 @@ public static void fireKeyInput(int key, int scanCode, int action, int modifiers
     {
         MinecraftForge.EVENT_BUS.post(new InputEvent.KeyInputEvent(key, scanCode, action, modifiers));
     }
-}
+    
+    /**
+     * Fixes MC-96521 directly, and marks the current world to be deleted by 
+     * {@link #deleteMarkedHardcoreWorld(WorldSummary)}
+     * 
+     * @see net.minecraft.client.gui.GuiGameOver
+     * @see #deleteMarkedHardcoreWorld
+     */
+    public static void fixHardcore() 
+    {
+        Minecraft mc = Minecraft.getInstance();
+        IntegratedServer server = ((IntegratedServer) ServerLifecycleHooks.getCurrentServer());
+        if(server != null) {
+	        server.addScheduledTask(() -> {
+	            String ownerName = server.getServerOwner();
+	            EntityPlayerMP owner = server.getPlayerList().getPlayerByUsername(ownerName);
+	            owner.setGameType(GameType.NOT_SET);
+	            mc.world.sendQuittingDisconnectingPacket();
```

Comment:
is this a race calling client code from server thread? should the server kick the client instead?

Reply:
Thanks! Don't know how I missed that, fixed.